src/tools/cleaner.js

```diff
@@ -30,27 +30,25 @@ const recurseCleanFuncs = (obj, path) => {
 };
 
 // Recurse a nested object replace all instances of keys->vals in the bank.
-const recurseReplaceBank = (obj, bank) => {
-  bank = bank || {};
+const recurseReplaceBank = (obj, bank = {}) => {
   const replacer = out => {
     if (typeof out !== 'string') {
       return out;
     }
     Object.keys(bank).forEach(key => {
+      // Escape characters (ex. {{foo}} => \\{\\{foo\\}\\} )
       const s = String(key).replace(/[-[\]/{}()\\*+?.^$|]/g, '\\$&');
       const re = new RegExp(s, 'g');
       out = out.replace(re, bank[key]);
     });
+
     return out;
   };
   return recurseReplace(obj, replacer);
 };
 
 // Takes a raw app and bundle and composes a bank of {{key}}->val
-const createBundleBank = (appRaw, event) => {
-  appRaw = appRaw || {};
-  event = event || {};
-
+const createBundleBank = (appRaw, event = {}) => {
```

Comment:
Mutable default arguments can lead to undesired effects when the function changes the argument, even though it doesn't seem to happen to this function. This [doc](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) explains why. It talks about Python, but it also applies to JavaScript.

Reply:
The change made is just syntax sugar In JS. Converting the linked example in Python to JS,

```
const append_to = (element, to=[]) => {
    to.push(element)
    return to;
}

const my_list = append_to(12)
const my_other_list = append_to(42)

console.log(my_list, my_other_list);
>>  [12], [42]
```

Could indeed be a little surprising coming from other languages.

Reply:
@szchenghuang thanks! TIL, this is indeed surprising!

Reply:
And this was a Python TIL for me @eliangcs 😂